sda/processor.go

```diff
@@ -124,10 +124,17 @@ func NewServiceProcessor(c *Context) *ServiceProcessor {
 	}
 }
 
-// RegisterMessage puts a new message in the message-handler
-// XXX More comments are needed as it's not clear whether RegisterMessage waits
-// for message for/from Clients or for/from Services.
-func (p *ServiceProcessor) RegisterMessage(f interface{}) error {
+// RegisterHandler will store the given handler that will be used by the service.
+// f must be a function that takes two input:
+//  * network.ServerIdentity: from whom the message is coming from
+//  * Pointer to a struct: message that the service is ready to handle.
+// f must have two return values:
+//  * Pointer to a struct: message that the service has generated as a reply and
+//  that will be sent to the requester (the sender).
+//  * Error in any case there is an error.
+// f can be used to treat internal service messages as well as external requests
```

Comment:
Still I don't think that is very clear. The only "thing" which want to keep general, is the 2nd argument of the passed (callback)-function. This could instead look sth like this:
`f func(sId *network.ServerIdentity, structPtr interface{})(network.Body, error)` instead of `f interface{}` and checking all properties of the passed argument during runtime. But this is orthogonal to this issue.